python/lsst/pipe/tasks/insertFakes.py

```diff
@@ -0,0 +1,572 @@
+        dimensions=("Tract", "Patch", "AbstractFilter", "SkyMap")
+    )
+
+    fakeCat = pipeBase.InputDatasetField(
+        doc="Catalog of fake sources to draw inputs from.",
+        nameTemplate="{CoaddName}Coadd_fakeSourceCat",
+        scalar=True,
+        storageClass="Parquet",
+        dimensions=("Tract", "SkyMap")
+    )
+
+    imageWithFakes = pipeBase.OutputDatasetField(
+        doc="Image with fake sources added.",
+        nameTemplate="fakes_{CoaddName}Coadd",
+        scalar=True,
+        storageClass="ExposureF",
+        dimensions=("Tract", "Patch", "AbstractFilter", "SkyMap")
+    )
+
+    coaddName = pexConfig.Field(
+        doc="The name of the type of coadd used",
+        dtype=str,
+        default="deep",
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.quantum.dimensions = ("Tract", "Patch", "AbstractFilter", "SkyMap")
+        self.formatTemplateNames({"CoaddName": "deep"})
+
+
+class InsertFakesTask(PipelineTask, CmdLineTask):
+    """Insert fake objects into images.
+
+    Add fake stars and galaxies to the given image, read in through the dataRef. Galaxy parameters are read in
+    from the specified file and then modelled using galsim.
+
+    `InsertFakesTask` has five functions that make images of the fake sources and then add them to the
+    image.
+
+    `addPixCoords`
+        Use the WCS information to add the pixel coordinates of each source.
+    `mkFakeGalsimGalaxies`
+        Use Galsim to make fake double sersic galaxies for each set of galaxy parameters in the input file.
+    `mkFakeStars`
+        Use the PSF information from the image to make a fake star using the magnitude information from the
+        input file.
+    `cleanCat`
+        Remove rows of the input fake catalog which have half light radius, of either the bulge or the disk,
+        that are 0.
+    `addFakeSources`
+        Add the fake sources to the image.
+
+    """
+
+    _DefaultName = "insertFakes"
+    ConfigClass = InsertFakesConfig
+
+    def runDataRef(self, dataRef):
+        """Read in/write out the required data products and add fake sources to the deepCoadd.
+
+        Parameters
+        ----------
+        dataRef : `lsst.daf.persistence.butlerSubset.ButlerDataRef`
+            Data reference defining the image to have fakes added to it
+            Used to access the following data products:
+                deepCoadd
+        """
+
+        # To do: should it warn when asked to insert variable sources into the coadd
+
+        if self.config.fakeType == "static":
+            fakeCat = dataRef.get("deepCoadd_fakeSourceCat").toDataFrame()
+            # To do: DM-16254, the read and write of the fake catalogs will be changed once the new pipeline
+            # task structure for ref cats is in place.
+            self.fakeSourceCatType = "deepCoadd_fakeSourceCat"
+        else:
+            fakeCat = Table.read(self.config.fakeType).to_pandas()
```

Comment:
Sorry to butt in uninvited:

Should this be reading `self.config.fakeCat`?  It's not obvious to me that `self.config.fakeType` will point to something that `Table` knows how to read.